src/Extension.Tests/QuestionTextEvaluationTests.cs

```diff
@@ -0,0 +1,145 @@
+        {
+            var evaluator = new Evaluator();
+            using var csharpKernel = new CSharpKernel();
+            using var kernel = new CompositeKernel { csharpKernel };
+            kernel
+                .UseQuestionMagicCommand(evaluator)
+                .UseEvaluateMagicCommand(evaluator);
+
+            using var events = kernel.KernelEvents.ToSubscribedList();
+
+            evaluator.AddQuestionTextCriterion("1", s => s.Contains("3.14"));
+
+            await kernel.SubmitCodeAsync(@"
+#!markdown
+#!question 1
+3.13
+");
+            events.Should().ContainSingle<QuestionTextEvaluationProduced>()
+                .Which.Evaluation.Passed.Should().Be(false);
+        }
+
+        [Fact]
+        public async Task submitting_text_that_passes_evaluation_criterion_gives_passed_evaluation()
+        {
+            var evaluator = new Evaluator();
+            using var csharpKernel = new CSharpKernel();
+            using var kernel = new CompositeKernel { csharpKernel };
+            kernel
+                .UseQuestionMagicCommand(evaluator)
+                .UseEvaluateMagicCommand(evaluator);
+
+            using var events = kernel.KernelEvents.ToSubscribedList();
+
+            evaluator.AddQuestionTextCriterion("1", s => s.Contains("3.14"));
+
+            await kernel.SubmitCodeAsync(@"
+#!markdown
+#!question 1
+3.14
+");
+            events.Should().ContainSingle<QuestionTextEvaluationProduced>()
+                .Which.Evaluation.Passed.Should().Be(true);
+        }
+
+
+        [Fact]
+        public async Task submitting_text_without_setting_evaluation_criterion_gives_passed_evaluation()
+        {
+            var evaluator = new Evaluator();
+            using var csharpKernel = new CSharpKernel();
+            using var kernel = new CompositeKernel { csharpKernel };
+            kernel
+                .UseQuestionMagicCommand(evaluator)
+                .UseEvaluateMagicCommand(evaluator);
+
+            using var events = kernel.KernelEvents.ToSubscribedList();
+
+            await kernel.SubmitCodeAsync(@"
+#!markdown
+#!question 1
+3.14
+");
+
+            events.Should().ContainSingle<QuestionTextEvaluationProduced>()
+                .Which.Evaluation.Passed.Should().Be(true);
+        }
+
+
+        [Fact]
+        public async Task multiple_evaluation_criteria_is_matched_to_the_correct_question()
+        {
+            var evaluator = new Evaluator();
+            using var csharpKernel = new CSharpKernel();
+            using var kernel = new CompositeKernel { csharpKernel };
+            kernel
+                .UseQuestionMagicCommand(evaluator)
+                .UseEvaluateMagicCommand(evaluator);
+
+            evaluator.AddQuestionTextCriterion("1", s => s.Contains("3.14"));
+            evaluator.AddQuestionTextCriterion("2", s => s.Contains("2.72"));
+
+            var resultQuestion2 = await kernel.SubmitCodeAsync(@"
+#!markdown
```

Comment:
Why is `#!markdown` being used here?

Reply:
This isn't necessary. But I thought adding this would stop KernelEvent from having CommandFailed. I felt it seemed weird if the tests intentionally caused commandfailed.

Reply:
Why is there a `CommandFailed`? Is that intended?

Reply:
This particular use case is where prof can use #!markdown cells to solicit plain text answers. Anything can be typed including strings that won't compile. If there wasn't #!markdown, it would give commandfailed due to syntax error.

I thought this use case would be part of input evaluation.

Reply:
This sounds like an increase in scope. We should avoid adding scenarios until the core ones are working.

Reply:
Got it. I'll remove it from this pr

Reply:
Done

Reply:
Often what I do in this kind of case is open an issue so we can discuss it later without adding complexity and additional design decisions into the current iteration.

Reply:
Sounds good. See #32